samples/point_sprites.cpp

```diff
@@ -0,0 +1,185 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <filament/Engine.h>
+#include <filament/IndexBuffer.h>
+#include <filament/Material.h>
+#include <filament/MaterialInstance.h>
+#include <filament/RenderableManager.h>
+#include <filament/Scene.h>
+#include <filament/TransformManager.h>
+#include <filament/VertexBuffer.h>
+#include <filament/View.h>
+
+#include <image/ImageSampler.h>
+#include <image/LinearImage.h>
+
+#include "../samples/app/Config.h"
+#include "../samples/app/FilamentApp.h"
+
+#include <cmath>
+
+#include "generated/resources/resources.h"
+
+using namespace filament;
+
+using utils::Entity;
+using utils::EntityManager;
+
+using MinFilter = TextureSampler::MinFilter;
+using MagFilter = TextureSampler::MagFilter;
+
+struct App {
+    VertexBuffer* vb;
+    IndexBuffer* ib;
+    Material* mat;
+    MaterialInstance* matInstance;
+    Camera* cam;
+    Texture* tex;
+    Entity renderable;
+};
+
+struct Vertex {
+    filament::math::float2 position;
+    uint32_t color;
+};
+
+#define NUM_POINTS 100
+#define TEXTURE_SIZE 128
+#define MAX_POINT_SIZE 128.0f
+
+void createSplatTexture(App& app, Engine* engine) {
+    static image::LinearImage splat(3, 3, 1);
+    splat.getPixelRef(1, 1)[0] = 1.0f;
+    splat = image::resampleImage(splat, TEXTURE_SIZE, TEXTURE_SIZE, image::Filter::GAUSSIAN_SCALARS);
+
+    Texture::PixelBufferDescriptor buffer(splat.getPixelRef(),
+            size_t(TEXTURE_SIZE * TEXTURE_SIZE * 4), Texture::Format::R, Texture::Type::FLOAT);
+
+    app.tex = Texture::Builder()
+            .width(TEXTURE_SIZE).height(TEXTURE_SIZE).levels(1)
+            .sampler(Texture::Sampler::SAMPLER_2D).format(Texture::InternalFormat::R32F)
+            .build(*engine);
+
+    app.tex->setImage(*engine, 0, std::move(buffer));
+}
+
+void setup(App& app, Engine* engine, View* view, Scene* scene) {
+
+    createSplatTexture(app, engine);
+
+    static Vertex kVertices[NUM_POINTS];
+    static float kPointSizes[NUM_POINTS];
+    static uint16_t kIndices[NUM_POINTS];
+
+    constexpr float dtheta = M_PI * 2 / NUM_POINTS;
+    constexpr float dsize = MAX_POINT_SIZE / NUM_POINTS;
+    constexpr float dcolor = 256.0f / NUM_POINTS;
+
+    for (int i = 0; i < NUM_POINTS; i++) {
+        const float theta = dtheta * i;
+        const uint32_t c = dcolor * i;
+        kVertices[i].position.x = cos(theta);
+        kVertices[i].position.y = sin(theta);
+        kVertices[i].color = 0xff000000u | c | (c << 8u) | (c << 16u);
+        kPointSizes[i] = 12.0f + dsize * i;
+        kIndices[i] = i;
+    }
+
+    view->setClearColor({0.1, 0.125, 0.25, 1.0});
+    view->setPostProcessingEnabled(false);
+    view->setDepthPrepass(filament::View::DepthPrepass::DISABLED);
+
+    app.vb = VertexBuffer::Builder()
```

Comment:
Same `sizeof(float)` suggestion here with all the multiples of 4

Reply:
Done.